packages/plugin-multi-tenant/src/types.ts

```diff
@@ -86,6 +98,8 @@ export type MultiTenantPluginConfig<ConfigTypes = unknown> = {
       }
     | {
         arrayFieldAccess?: never
+        arrayFieldName?: never
+        arrayTenantFieldName?: never
```

Comment:
These actually should be the same type as in the other union here so if you add your own field the plugin still has knowledge of the field names.

so,
```ts
  arrayFieldName?: string
  arrayTenantFieldName?: string
```